src/duckula/avro/schema.clj

```diff
@@ -34,27 +37,37 @@
     ;; Complex Types
     Schema$Type/RECORD
     (apply hash-map (mapcat (fn [key val]
-                              [(keyword key) val])
-                            (map #(.name %) (.getFields avro-schema))
-                            (map #(->map (.schema %)) (.getFields avro-schema))))
+                              (let [k (if mangle-names?
+                                        (-> key (str/replace \_ \-) keyword)
+                                        (keyword key))]
+                                [k val]))
```

Comment:
no need for the let:

```clj
[(if mangle-names?
    (-> key (str/replace \_ \-) keyword)
    (keyword key))
  val]
```

That said, this function is still a bit hard to follow

Reply:
I can extract bits of it, but it will largely be unchanged - it's recursive in nature, so not much I can do about, even after we skip the step of generating Prismatic Schema.

Regarding the suggestion - that was the code I had originally, I found it harder to read because of the `if` being inside of a vector.